### DISCUSSION TITLE
Cypress tests failing in firefox in pages with polling requests

### DISCUSSION THREAD

**Discussion by Ollisteka** [cypress-io/cypress#19865] [Questions and Help]
I have a following test in Cypress:
- visit first page with the header `A`
- click on the `Go to B Page` button
- assert that the header of the page is now `B`

It works fine in Chrome, but failing in Firefox, as on the page `B` I have some background polling requests, and when cypress switches to another test and those requests get "canceled" away, I get either `TypeError: NetworkError when attempting to fetch resource` or `AbortError: The operation was aborted`

All the requests are using fetch api, by the way.

The possibility to mute those errors through the `uncaught:exception` event seems a bad idea, and so does the idea to do something on the page to cancel the polling, as it is not the thing under testing.

Maybe someone has encoutnered this problem too and got some non-hacky solution?

**Comment by georgekaran** [comment]
Hey @Ollisteka

Did you manage to find a solution for this problem?

**Reply by Ollisteka** [reply]
Nope! I muted the errors through the `uncaught:exception` 🙄

```js

/**
* If we leave the page where there were hanging requests, Firefox throws an error, which causes the tests to fail.
 *
 * https://github.com/cypress-io/cypress/discussions/19865
 */
const ignoreFirefoxHangingRequestError = () => {
    // I use Cypress.on instead of cy.on because cy.on does not catch exceptions inside beforeEach
    Cypress.on('uncaught:exception', e => {
        if (
            Cypress.browser.name === 'firefox' &&
            (e.message.includes('AbortError') || e.message.includes('NetworkError'))
        ) {
            return false;
        }
    });
};

ignoreFirefoxHangingRequestError();
```

**Reply by georgekaran** [reply]
Yeah 😞 

It's either this or controlling the polling requests via environment variables, example, if you're using a setTimeout, in the testing environment you could set a really high number to avoid this polling to be called and interfere in your tests.

**Reply by Ollisteka** [reply]
Modifying production code, even through the environment variables sounds as the worse option for me — if the code does not check for the browser name, and just diables/delays the request, then it means that the functionality is not tested at all. And what if there is a bug? So better this code will be tested at chrome only, than nowhere at all

Also, modifying the production code kind of ruins the idea of testing your code as the black box)
